diff -Naur a/libredirect.c b/libredirect.c
--- a/libredirect.c	2016-08-12 15:17:45.775206601 +0200
+++ b/libredirect.c	2016-08-12 15:21:18.747316335 +0200
@@ -24,8 +24,6 @@
     char * spec = getenv("NIX_REDIRECTS");
     if (!spec) return;
 
-    unsetenv("NIX_REDIRECTS");
-
     char * spec2 = malloc(strlen(spec) + 1);
     strcpy(spec2, spec);
 
@@ -118,13 +116,34 @@
     return __xstat64_real(ver, rewrite(path, buf), st);
 }
 
-int * access(const char * path, int mode)
+int __lxstat(int ver, const char * path, struct stat * st)
 {
-    int * (*access_real) (const char *, int mode) = dlsym(RTLD_NEXT, "access");
+    int (*__lxstat_real) (int ver, const char *, struct stat *) = dlsym(RTLD_NEXT, "__lxstat");
+    char buf[PATH_MAX];
+    return __lxstat_real(ver, rewrite(path, buf), st);
+}
+
+int __lxstat64(int ver, const char * path, struct stat64 * st)
+{
+    int (*__lxstat64_real) (int ver, const char *, struct stat64 *) = dlsym(RTLD_NEXT, "__lxstat64");
+    char buf[PATH_MAX];
+    return __lxstat64_real(ver, rewrite(path, buf), st);
+}
+
+int access(const char * path, int mode)
+{
+    int (*access_real) (const char *, int mode) = dlsym(RTLD_NEXT, "access");
     char buf[PATH_MAX];
     return access_real(rewrite(path, buf), mode);
 }
 
+int eaccess(const char * path, int mode)
+{
+    int (*eaccess_real) (const char *, int mode) = dlsym(RTLD_NEXT, "eaccess");
+    char buf[PATH_MAX];
+    return eaccess_real(rewrite(path, buf), mode);
+}
+
 int posix_spawn(pid_t * pid, const char * path,
     const posix_spawn_file_actions_t * file_actions,
     const posix_spawnattr_t * attrp,
