diff --git a/src/business/business-ledger/gncEntryLedger.c b/src/business/business-ledger/gncEntryLedger.c
index f2f28b2a4..e12025210 100644
--- a/src/business/business-ledger/gncEntryLedger.c
+++ b/src/business/business-ledger/gncEntryLedger.c
@@ -260,21 +260,21 @@ gnc_entry_ledger_config_cells (GncEntryLedger *ledger)
     ((ComboCell *)
      gnc_table_layout_get_cell (ledger->table->layout, ENTRY_ACTN_CELL), FALSE);
 
-    /* Use 6 decimal places for all prices and quantities */
+    /* Use 8 decimal places for all prices and quantities */
     gnc_price_cell_set_fraction
     ((PriceCell *)
      gnc_table_layout_get_cell (ledger->table->layout, ENTRY_PRIC_CELL),
-     1000000);
+     100000000);
 
     gnc_price_cell_set_fraction
     ((PriceCell *)
      gnc_table_layout_get_cell (ledger->table->layout, ENTRY_DISC_CELL),
-     1000000);
+     100000000);
 
     gnc_price_cell_set_fraction
     ((PriceCell *) gnc_table_layout_get_cell (ledger->table->layout,
             ENTRY_QTY_CELL),
-     1000000);
+     100000000);
 
     /* add menu items for the action and payment cells */
     gnc_entry_ledger_config_action (ledger);
diff --git a/src/engine/Account.c b/src/engine/Account.c
index 9a4e3965f..458b341ff 100644
--- a/src/engine/Account.c
+++ b/src/engine/Account.c
@@ -609,7 +609,7 @@ gnc_account_class_init (AccountClass *klass)
                        "1 USD can be divided into 100 cents.",
                        0,
                        G_MAXINT32,
-                       1000000,
+                       100000000,
                        G_PARAM_READWRITE));
 
     g_object_class_install_property
diff --git a/src/engine/Split.c b/src/engine/Split.c
index 6ed3a4ebf..f8f86c596 100644
--- a/src/engine/Split.c
+++ b/src/engine/Split.c
@@ -928,7 +928,7 @@ get_currency_denom(const Split * s)
     }
     else if (!s->parent || !s->parent->common_currency)
     {
-        return 1000000; /* Max supported denom to avoid premature rounding. */
+        return 100000000; /* Max supported denom to avoid premature rounding. */
     }
     else
     {
@@ -945,7 +945,7 @@ get_commodity_denom(const Split * s)
     }
     else if (!s->acc)
     {
-        return 1000000; /* Max supported denom to avoid premature rounding. */
+        return 100000000; /* Max supported denom to avoid premature rounding. */
     }
     else
     {
diff --git a/src/engine/gnc-commodity.c b/src/engine/gnc-commodity.c
index a48cb793b..a38606da1 100644
--- a/src/engine/gnc-commodity.c
+++ b/src/engine/gnc-commodity.c
@@ -801,7 +801,7 @@ gnc_commodity_class_init(struct _GncCommodityClass* klass)
                                             "The fraction is the number of sub-units that "
                                             "the basic commodity can be divided into.",
                                             1,
-                                            1000000,
+                                            100000000,
                                             1,
                                             G_PARAM_READWRITE));
     g_object_class_install_property(gobject_class,
diff --git a/src/engine/iso-4217-currencies.xml b/src/engine/iso-4217-currencies.xml
index 24429728e..b442c10c6 100644
--- a/src/engine/iso-4217-currencies.xml
+++ b/src/engine/iso-4217-currencies.xml
@@ -3036,4 +3036,38 @@
   smallest-fraction="1"
   local-symbol=""
 />
+<!-- cryptos follow -->
+<currency
+  isocode="BTC"
+  fullname="Bitcoin"
+  unitname="BTC"
+  partname="mBTC"
+  namespace="ISO4217"
+  exchange-code="5001"
+  parts-per-unit="1000"
+  smallest-fraction="100000000"
+  local-symbol=""
+/>
+<currency
+  isocode="ETH"
+  fullname="Ethereum"
+  unitname="ETH"
+  partname="mETH"
+  namespace="ISO4217"
+  exchange-code="5002"
+  parts-per-unit="1000"
+  smallest-fraction="100000000"
+  local-symbol=""
+/>
+<currency
+  isocode="ADA"
+  fullname="Ada"
+  unitname="ADA"
+  partname="mADA"
+  namespace="ISO4217"
+  exchange-code="5003"
+  parts-per-unit="1000"
+  smallest-fraction="100000000"
+  local-symbol=""
+/>
 </currencylist>
diff --git a/src/engine/test/utest-Split.c b/src/engine/test/utest-Split.c
index 64a2a60fc..4300631cc 100644
--- a/src/engine/test/utest-Split.c
+++ b/src/engine/test/utest-Split.c
@@ -748,10 +748,10 @@ test_get_currency_denom (Fixture *fixture, gconstpointer pData)
     const gint denom = gnc_commodity_get_fraction (fixture->curr);
     g_assert_cmpint (fixture->func->get_currency_denom (NULL), ==, 0);
     fixture->split->parent = NULL;
-    g_assert_cmpint (fixture->func->get_currency_denom (fixture->split), ==, 1000000);
+    g_assert_cmpint (fixture->func->get_currency_denom (fixture->split), ==, 100000000);
     fixture->split->parent = txn;
     txn->common_currency = NULL;
-    g_assert_cmpint (fixture->func->get_currency_denom (fixture->split), ==, 1000000);
+    g_assert_cmpint (fixture->func->get_currency_denom (fixture->split), ==, 100000000);
     txn->common_currency = fixture->curr;
     g_assert_cmpint (fixture->func->get_currency_denom (fixture->split), ==, denom);
 }
@@ -766,7 +766,7 @@ test_get_commodity_denom (Fixture *fixture, gconstpointer pData)
     const gint denom = gnc_commodity_get_fraction (fixture->comm);
     g_assert_cmpint (fixture->func->get_commodity_denom (NULL), ==, 0);
     fixture->split->acc = NULL;
-    g_assert_cmpint (fixture->func->get_commodity_denom (fixture->split), ==, 1000000);
+    g_assert_cmpint (fixture->func->get_commodity_denom (fixture->split), ==, 100000000);
     fixture->split->acc = acc;
     g_assert_cmpint (fixture->func->get_commodity_denom (fixture->split), ==, denom);
 }
diff --git a/src/gnome-utils/gnc-tree-view-split-reg.c b/src/gnome-utils/gnc-tree-view-split-reg.c
index 08588b284..8433e2733 100644
--- a/src/gnome-utils/gnc-tree-view-split-reg.c
+++ b/src/gnome-utils/gnc-tree-view-split-reg.c
@@ -1841,7 +1841,7 @@ gtv_sr_cdf0 (GtkTreeViewColumn *col, GtkCellRenderer *cell, GtkTreeModel *s_mode
             print_info = gnc_default_price_print_info();
             print_info.min_decimal_places = 2;
 
-            num = gnc_numeric_convert (gnc_tree_util_get_rate_for (view, trans, split, is_blank), 1000000, GNC_HOW_RND_ROUND_HALF_UP);
+            num = gnc_numeric_convert (gnc_tree_util_get_rate_for (view, trans, split, is_blank), 100000000, GNC_HOW_RND_ROUND_HALF_UP);
 
             if (gnc_numeric_check (num) == GNC_ERROR_OK)
                 s = xaccPrintAmount (num, print_info);
diff --git a/src/gnome-utils/gtkbuilder/dialog-account.glade b/src/gnome-utils/gtkbuilder/dialog-account.glade
index 82d7f1aaa..e66a80c39 100644
--- a/src/gnome-utils/gtkbuilder/dialog-account.glade
+++ b/src/gnome-utils/gtkbuilder/dialog-account.glade
@@ -32,6 +32,12 @@
       <row>
         <col id="0" translatable="yes">1/1000000</col>
       </row>
+      <row>
+        <col id="0" translatable="yes">1/10000000</col>
+      </row>
+      <row>
+        <col id="0" translatable="yes">1/100000000</col>
+      </row>
     </data>
   </object>
   <object class="GtkDialog" id="Account Dialog">
diff --git a/src/register/ledger-core/split-register-util.c b/src/register/ledger-core/split-register-util.c
index 2211d222b..f1fec0584 100644
--- a/src/register/ledger-core/split-register-util.c
+++ b/src/register/ledger-core/split-register-util.c
@@ -426,7 +426,7 @@ gnc_split_register_set_cell_fractions (SplitRegister *reg, Split *split)
     if (split_account)
         gnc_price_cell_set_fraction (cell, xaccAccountGetCommoditySCU (split_account));
     else
-        gnc_price_cell_set_fraction (cell, 1000000);
+        gnc_price_cell_set_fraction (cell, 100000000);
 }
 
 CellBlock *
diff --git a/src/register/ledger-core/split-register.c b/src/register/ledger-core/split-register.c
index 2e2aebdee..62453c582 100644
--- a/src/register/ledger-core/split-register.c
+++ b/src/register/ledger-core/split-register.c
@@ -66,7 +66,7 @@ static SCM copied_item = SCM_UNDEFINED;
 static GncGUID copied_leader_guid;
 /* A denominator representing number of digits to the right of the decimal point
  * displayed in a price cell. */
-static int PRICE_CELL_DENOM = 1000000;
+static int PRICE_CELL_DENOM = 100000000;
 /** static prototypes *****************************************************/
 
 static gboolean gnc_split_register_save_to_scm (SplitRegister *reg,
@@ -2597,7 +2597,7 @@ gnc_split_register_config_cells (SplitRegister *reg)
     ((ComboCell *)
      gnc_table_layout_get_cell (reg->table->layout, ACTN_CELL), TRUE);
 
-    /* Use 6 decimal places for prices and "exchange rates"  */
+    /* Use 8 decimal places for prices and "exchange rates"  */
     gnc_price_cell_set_fraction
     ((PriceCell *)
      gnc_table_layout_get_cell (reg->table->layout, PRIC_CELL),
