diff --git a/app/Main.hs b/app/Main.hs
index 5811ff5..339c1f7 100644
--- a/app/Main.hs
+++ b/app/Main.hs
@@ -14,6 +14,7 @@ import Control.Concurrent.STM.TMVar
 import Control.Concurrent.STM.TQueue.Extra
 import Control.Exception
 import Control.Monad
+import Control.Monad.Catch (MonadThrow(..))
 import Control.Monad.Managed
 import Data.ByteString (ByteString)
 import Data.List.NonEmpty (NonEmpty(..))
diff --git a/steeloverseer.cabal b/steeloverseer.cabal
index e6bea6c..63e6a82 100644
--- a/steeloverseer.cabal
+++ b/steeloverseer.cabal
@@ -47,13 +47,13 @@ library
         , regex-tdfa >= 1.2
         , semigroups >= 0.16
         , stm >= 2.4
-        , streaming >= 0.1.0 && < 0.1.5
+        , streaming
         , text >= 1.2
         , yaml >= 0.8
         , aeson         >= 0.8
         , ansi-terminal >= 0.6.2
         , containers    >= 0.5
-        , process       >= 1.6.0 && < 1.6.1
+        , process
         , unix
     hs-source-dirs:
           src
@@ -73,7 +73,7 @@ executable sos
         , regex-tdfa >= 1.2
         , semigroups >= 0.16
         , stm >= 2.4
-        , streaming >= 0.1.0 && < 0.1.5
+        , streaming
         , text >= 1.2
         , yaml >= 0.8
         , steeloverseer
@@ -107,7 +107,7 @@ test-suite spec
         , regex-tdfa >= 1.2
         , semigroups >= 0.16
         , stm >= 2.4
-        , streaming >= 0.1.0 && < 0.1.5
+        , streaming
         , text >= 1.2
         , yaml >= 0.8
         , steeloverseer
