diff --git a/modules/clicks-in-new-buffer.js b/modules/clicks-in-new-buffer.js
index 40b5f99..f606f60 100644
--- a/modules/clicks-in-new-buffer.js
+++ b/modules/clicks-in-new-buffer.js
@@ -32,7 +32,7 @@ function find_tag_in_parents (tag, element) {
 }
 
 function open_link_in_new_buffer (event) {
-    if (event.button != clicks_in_new_buffer_button)
+    if (event.button != clicks_in_new_buffer_button && ! event.ctrlKey)
         return;
     let element = event.target;
     let anchor = null;
