diff --git a/libtransmission/torrent.c b/libtransmission/torrent.c
index 7ecaf3f51..6f6a07a3e 100644
--- a/libtransmission/torrent.c
+++ b/libtransmission/torrent.c
@@ -3063,6 +3063,8 @@ static void deleteLocalData(tr_torrent* tor, tr_fileFunc func)
     PtrArrayCompareFunc vstrcmp = (PtrArrayCompareFunc)strcmp;
     char const* const top = tor->currentDir;
 
+    return;
+
     /* don't try to delete local data if the directory's gone missing */
     if (!tr_sys_path_exists(top, NULL))
     {
diff --git a/web/index.html b/web/index.html
index 6c9518a29..7b4c11f56 100755
--- a/web/index.html
+++ b/web/index.html
@@ -507,7 +507,6 @@
 			<li data-command="move_bottom">Move to Bottom</li>
 			<li>---</li>
 			<li data-command="remove">Remove From List…</li>
-			<li data-command="remove_data">Trash Data and Remove From List…</li>
 			<li>---</li>
 			<li data-command="verify">Verify Local Data</li>
 			<li data-command="move">Set Location…</li>
