diff --git a/st.c b/st.c
index 124ed16..107e02f 100644
--- a/st.c
+++ b/st.c
@@ -2697,7 +2697,7 @@ externalpipe(const Arg *arg)
 	if(pipe(to) == -1)
 		return;
 
-	/* sigchld() handles this */
+	/* sigchld() handles this — <michalrus> no, it doesn’t… */
 	switch(child = fork()){
 		case -1:
 			close(to[0]), close(to[1]);
@@ -2738,6 +2738,9 @@ externalpipe(const Arg *arg)
 
 	/* restore */
 	signal(SIGPIPE, oldsigpipe);
+
+	/* <michalrus> let’s block here to avoid zombie processes… */
+	waitpid(child, NULL, 0);
 }
 
 void
