diff --git a/config.def.h b/config.def.h
index 64e75b8..d21aa62 100644
--- a/config.def.h
+++ b/config.def.h
@@ -62,29 +62,30 @@ static unsigned int tabspaces = 8;
 /* Terminal colors (16 first used in escape sequence) */
 static const char *colorname[] = {
 	/* 8 normal colors */
-	"black",
-	"red3",
-	"green3",
-	"yellow3",
-	"blue2",
-	"magenta3",
-	"cyan3",
-	"gray90",
+	"#073642",  /*  0: black    */
+	"#dc322f",  /*  1: red      */
+	"#859900",  /*  2: green    */
+	"#b58900",  /*  3: yellow   */
+	"#268bd2",  /*  4: blue     */
+	"#d33682",  /*  5: magenta  */
+	"#2aa198",  /*  6: cyan     */
+	"#93a1a1",  /*  7: white    */

 	/* 8 bright colors */
-	"gray50",
-	"red",
-	"green",
-	"yellow",
-	"#5c5cff",
-	"magenta",
-	"cyan",
-	"white",
+	"#073642",  /*  8: brblack  */
+	"#cb4b16",  /*  9: brred    */
+	"#859900",  /* 10: brgreen  */
+	"#b58900",  /* 11: bryellow */
+	"#268bd2",  /* 12: brblue   */
+	"#d33682",  /* 13: brmagenta*/
+	"#2aa198",  /* 14: brcyan   */
+	"#fdf6e3",  /* 15: brwhite  */

 	[255] = 0,

 	/* more colors can be added after 255 to use with DefaultXX */
-	"#cccccc",
+	"#93a1a1",
+	"#002b36"
 };


@@ -92,8 +93,8 @@ static const char *colorname[] = {
  * Default colors (colorname index)
  * foreground, background, cursor
  */
-static unsigned int defaultfg = 7;
-static unsigned int defaultbg = 0;
+static unsigned int defaultfg = 256;
+static unsigned int defaultbg = 257;
 static unsigned int defaultcs = 256;

 /*
